package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,414 @@
+            # weights do not apply to metric but density of data
+            self._weights_ker = np.ones((self._nframes,))
+        else:
+            if weights.shape[0] != self._nframes:
+                raise ValueError("The weight should have the same length as "
+                                 'the trajectory')
+            else:
+                # weights are constructed as relative to the mean
+                self._weights_ker = (np.asarray(weights, dtype=np.float64) /
+                                     np.mean(weights))
+
+    def decompose_kernel(self):
+        # this should be a reference to the same object as
+        # self.DistanceMatrix.dist_matrix
+        self._kernel = self._epsilon.scaledMatrix
+
+        # take negative exponent of scaled matrix to create Isotropic kernel
+        self._kernel = np.exp(-self._kernel)
+
+        # define an anistropic diffusion term q
+        q_vector = np.zeros((self._nframes, ))
+        # weights should reflect the density of the points on the manifold
+        for i in range(self._nframes):
+            q_vector[i] = np.dot(self._kernel[i, :], self._weights_ker)
+
+        # Form a new kernel from the anisotropic diffusion term q
+        self._kernel /= np.sqrt(q_vector[:, np.newaxis].dot(q_vector[np.newaxis]))
+
+        # Weighted Graph Laplacian normalization on this new graph
+        d_vector = np.zeros((self._nframes, ))
+        for i in range(self._nframes):
+            d_vector[i] = np.dot(self._kernel[i, :], self._weights_ker)
+
+        for i in range(self._nframes):
+            self._kernel[i, :] = self._kernel[i, :] * self._weights_ker
+
+        # Define anisotropic transition by dividing kernel by this term
+        self._kernel /= np.sqrt(d_vector[:, np.newaxis].dot(d_vector[np.newaxis]))
+
+        # Apply timescaling
+        for i in range(self._t):
+            if i > 0:
+                self._kernel = self._kernel.dot(self._kernel.T)
+
+        eigenvals, eigenvectors = np.linalg.eig(self._kernel)
+
+        eg_arg = np.argsort(eigenvals)
+        self.eigenvalues = eigenvals[eg_arg[::-1]]
+        self.eigenvectors = eigenvectors[eg_arg[::-1], :]
+
+    def _prepare(self):
+        self.fit = np.zeros((self.eigenvectors.shape[0],
+                             self.num_eigenvectors))
+
+    def _single_frame(self):
+        # The diffusion map embedding takes the ith sample in the
+        # data matrix and maps it to each of the ith coordinates
+        # in the set of k-dominant eigenvectors
+        for k in range(self.num_eigenvectors):
+            self.fit[self._ts.frame][k] = self.eigenvectors[k][self._ts.frame]
+
+    def transform(self, num_eigenvectors):
+        """ Embeds a trajectory via the diffusion map
+
+        Parameter
+        ---------
+        num_eigenvectors : int
+            The number of dominant eigenvectors to be used for
+            diffusion mapping
+
+        Return
+        ------
+        fit : array
```

Comment:
why call it fit? you are not fitting? I don't think we need to save this extra in the class as well because it is just a *reshape* of the eigenvectors